federation/pkg/federation-controller/sync/controller.go

```diff
@@ -424,29 +475,54 @@ type clusterObjectAccessorFunc func(clusterName string) (interface{}, bool, erro
 // clusterOperations returns the list of operations needed to synchronize the state of the given object to the provided clusters
 func clusterOperations(adapter federatedtypes.FederatedTypeAdapter, clusters []*federationapi.Cluster, obj pkgruntime.Object, accessor clusterObjectAccessorFunc) ([]util.FederatedOperation, error) {
 	key := federatedtypes.ObjectKey(adapter, obj)
-	operations := make([]util.FederatedOperation, 0)
+	current := make(map[string]pkgruntime.Object)
 	for _, cluster := range clusters {
 		clusterObj, found, err := accessor(cluster.Name)
 		if err != nil {
 			return nil, fmt.Errorf("Failed to get %s %q from cluster %q: %v", adapter.Kind(), key, cluster.Name, err)
 		}
-		// The data should not be modified.
-		desiredObj := adapter.Copy(obj)
 
-		var operationType util.FederatedOperationType = ""
 		if found {
-			clusterObj := clusterObj.(pkgruntime.Object)
-			if !adapter.Equivalent(desiredObj, clusterObj) {
+			current[cluster.Name] = clusterObj.(pkgruntime.Object)
+		} else {
+			current[cluster.Name] = nil
+		}
+	}
+
+	scheduled := make(map[string]pkgruntime.Object)
```

Comment:
The (additional) Schedule () method is actually that function. The tests on the adapter implementation where I implement this method with test this path (Will submit that PR after something conclusive happens here.. :-)).